### ISSUE TITLE
No support on PE of 64 bit

### ISSUE THREAD

**Issue by t3mp1337** [zeroq/peanalysis#2]
When I am trying to use this project on 64 bit binaries it doesn't read them.
It gives me an empty imports table.

You need to fix the calculation of the imported table in 64 bit binaries.
Here is an example of the another python project that works for 32 and 64 bit files:
https://github.com/erocarrera/pefile/blob/master/pefile.py#L4024

I prefer yours because it works faster.

**Comment by t3mp1337** [comment]
I commit changes that support also PE of 64 bit.
My changes are on the optional header (added support for structure of 64 bit).
Works fine on the imports.
I didn't see other problems, so for now it good. Just accept my changes or change them a little.

**Comment by zeroq** [comment]
wow that was quick, I was just about to take a look at the 64Bit support. I'll take a look at your changes and add them as soon as possible

**Comment by t3mp1337** [comment]
It will be hard for you to see changes because I changed also the indentation.
So I will tell you what functions I changed:

**Comment by t3mp1337** [comment]
It will be hard for you to see changes because I changed also the indentation.
So I will tell you what I changed.


1.  In `def __init__(self, fn, content=None)` I added:
	`self.Is32bit = True`
2.  In the `readPEOptHeader(self, peOptionalHeader)` function I added new function named `fillOptHeaderFields(self, peOptionalHeader)`.
	This function checks if this is a 32 or 64 bit PE file.
	You handle 32 bit file so I added the case when this is a 64 bit file.
	In 64 bit file the IMAGE_OPTIONAL_HEADER is different.
	Therefore I changed some fields:
```
	# There is not 'baseofdata' in 64 bit so I made it zero.
	self.peoptDict['09_baseofdata'] = 0 

	# The following 5 fields are type of long long  ('q' instead of 'i') and they should take 8 bytes each.
	# So instead of 28:32 I changed to 24:32 and so on.
	self.peoptDict['10_imagebase'] = struct.unpack('q', peOptionalHeader[24:32])[0]
	self.peoptDict['25_SizeOfStackReserve'] = struct.unpack('q', peOptionalHeader[72:80])[0]
	self.peoptDict['26_SizeOfStackCommit'] = struct.unpack('q', peOptionalHeader[80:88])[0]
	self.peoptDict['27_SizeOfHeapReserve'] = struct.unpack('q', peOptionalHeader[88:96])[0]
	self.peoptDict['28_SizeOfHeapCommit'] = struct.unpack('q', peOptionalHeader[96:104])[0]
```

You made `init1 = 96` and `init2 = 100`.
	But when working with 64 bit it should be different.
	We read more bytes (+12 more than 64 bit). 
	So instead of `init1 = 96` on 32 bit, it will be `init1 = 112` on 64 bit.
	Init2 will be +4.

3.  In def getImportedFunctions(self, retVal=False):
    The fRVA calculation is different in 64 bit.
	You did:
	`fRVA = struct.unpack('I', self.secionDataDict[self.exportSymbolDict[item]['sectionID']][index:index+4])[0]`
	
	But in 64 bit you need to read 8 ('q', long long) bytes and not 4.
	And also the offset of the index should jump +8 and not +4.
	So I added simple check:
```
	if self.Is32bit:
            struct_unpack_fmt = 'I'
            index_offset = 4
        else:
            struct_unpack_fmt = 'q'
            index_offset = 8
```
		
And the fRVA calculation to:
		`fRVA = struct.unpack(struct_unpack_fmt, self.secionDataDict[self.importSymbolDict[item]['sectionID']][index:index + index_offset])[0]`
		
**I didn't do the changes on the export function.
I supposed we will also need to do such changes there.**

		
4.  Ordinal problem:
This is also in getImportedFunctions(self, retVal=False)
When I analyze files with imported functions that have ordinals it throw exception.
You did:
```
	if retVal:
		dllDict[currDLL].append("Ordinal: %s" % (int(hex(fRVA), 16) ^ int('0x80000000', 16)))
	else:
        print("\t\t load by ordinal: %s" % (int(hex(fRVA), 16) ^ int('0x80000000', 16)))
```
		
I changed it to 'long' instead of 'int':
```
	if retVal:
		dllDict[currDLL].append("Ordinal: %s" % (long(hex(fRVA), 16) ^ long('0x80000000', 16)))
	else:
        print("\t\t load by ordinal: %s" % (long(hex(fRVA), 16) ^ long('0x80000000', 16)))

```

**Comment by zeroq** [comment]
thank you very much, i have added your changes. One minor thing is the index += 4 on line 330, that needs to be index += index_offset then.
I also checked for the exports with a 64bit DLL from nvidia, but here everything seems to be fine.

**Comment by t3mp1337** [comment]
I also didn't find problem on the exports of 64 bit files.
But I think this is because the problem will happen when there are **exports with ordinal**.

If you will find 64 bit file with exports that contain ordinals, you will receive exception because of:
```
if hex(fRVA).startswith('0x8000'):
  if retVal:
       dllDict[currDLL].append("Ordinal: %s" % (int(hex(fRVA), 16) ^ int('0x80000000', 16)))
  else:
       print("\t\t load by ordinal: %s" % (int(hex(fRVA), 16) ^ int('0x80000000', 16)))
```

It similar to the changes we did in the imports.
Think about that. 
You are checking if fRVA starts with '0x8000' right ?
So it can be an adddress like: 2147483729L

If you will run this:
```
fRVA = 2147483729L
int(hex(fRVA), 16)
```

You will receive an exception.
I tried to search for exports on 64 bit with ordinal, but without success so I can't find a way to prove it.
The **only** change I would do here is from 'int' to 'long'.

The other things looks to work good so I wouldn't add changes like:
`fRVA = struct.unpack('Q'....` instead of `fRVA = struct.unpack('I'....` for 64 bit.
It looks to work ok without this change.
